File: frontend/src/types/request.ts
export type BudgetLevel = 'budget' | 'moderate' | 'luxury';
export type TravelPace = 'relaxed' | 'moderate' | 'intensive';
export type InterestTag =
  | 'temples' | 'food' | 'shopping' | 'nature' | 'anime'
  | 'history' | 'onsen' | 'nightlife' | 'photography' | 'family';

export interface TripRequest {
  start_date: string;
  end_date: string;
  cities: string[];
  budget_level: BudgetLevel;
  pace: TravelPace;
  interests: InterestTag[];
  include_jr_pass_calc: boolean;
  language: string;
}

export interface CityOption {
  id: string;
  name_en: string;
  name_ja: string;
  name_zh: string;
  region: string;
  recommended_days: { min: number; max: number };
}

File: backend/app/models/request.py
from datetime import date
from enum import Enum

from pydantic import BaseModel, Field


class BudgetLevel(str, Enum):
    BUDGET = "budget"
    MODERATE = "moderate"
    LUXURY = "luxury"


class TravelPace(str, Enum):
    RELAXED = "relaxed"
    MODERATE = "moderate"
    INTENSIVE = "intensive"


class InterestTag(str, Enum):
    TEMPLES = "temples"
    FOOD = "food"
    SHOPPING = "shopping"
    NATURE = "nature"
    ANIME = "anime"
    HISTORY = "history"
    ONSEN = "onsen"
    NIGHTLIFE = "nightlife"
    PHOTOGRAPHY = "photography"
    FAMILY = "family"


class TripRequest(BaseModel):
    start_date: date
    end_date: date
    cities: list[str] = Field(..., min_length=1, max_length=8)
    budget_level: BudgetLevel = BudgetLevel.MODERATE
    pace: TravelPace = TravelPace.MODERATE
    interests: list[InterestTag] = []
    include_jr_pass_calc: bool = True
    language: str = Field(default="zh", pattern="^(zh|en)$")

    @property
    def num_days(self) -> int:
        return (self.end_date - self.start_date).days + 1
